tunnel/dns/server.go

```diff
@@ -98,7 +101,14 @@ func NewDnsServer(addr string) Resolver {
 	}
 
 	names := make(map[string]net.IP)
-	r := &resolver{server: s, names: names, namesMtx: sync.Mutex{}}
+	r := &resolver{
+		server:     s,
+		names:      names,
+		ips:        make(map[string]string),
+		namesMtx:   sync.Mutex{},
+		domains:    make(map[string]*domainEntry),
+		domainsMtx: sync.Mutex{},
```

Comment:
Why this instead of a concurrent map? I assume to avoid having to deal with `interfac{}`